client/my-sites/checkout/upsell-nudge/professional-email-upsell/index.tsx

```diff
@@ -109,12 +120,11 @@ const ProfessionalEmailUpsell = ( {
 	 */
 	const getProratedPrice = ( productCost: number, freeMonths: number ) => {
 		const now = new Date();
-		const freeDays = new Date( new Date().setMonth( now.getMonth() + freeMonths ) ).getTime();
-		const nextYearDate = new Date( new Date().setFullYear( now.getFullYear() + 1 ) );
-		const diff = new Date( nextYearDate.getTime() - freeDays );
-		const diffDays =
-			( nextYearDate.getTime() - now.getTime() ) / 86400000 - diff.getTime() / 86400000;
-		const price = ( 365 - diffDays ) / 365;
+		const freeTime = addMonths( new Date(), freeMonths ).getTime();
+		const nextYearDate = addMonths( new Date(), 12 );
+		const diff = nextYearDate.getTime() - freeTime;
+		const diffDays = ( nextYearDate.getTime() - now.getTime() ) / 86400000 - diff / 86400000;
+		const price = ( 365 - Math.round( diffDays ) ) / 365;
```

Comment:
I think our billing engine may be using the equivalent of `Math.floor( diffDays )` to ensure we don't overcharge users, but I will dig in more deeply tomorrow morning.